modules/apps/info/info-impl/src/main/java/com/liferay/info/internal/item/InfoItemServiceTrackerImpl.java

```diff
@@ -279,9 +312,61 @@ private InfoItemClassDetails _getInfoItemClassDetails(
 
 	private ServiceTrackerMap<String, InfoItemCapability>
 		_infoItemCapabilityServiceTrackerMap;
-	private final Map<String, ServiceTrackerMap<String, ? extends List<?>>>
-		_itemClassNameInfoItemServiceTrackerMap = new HashMap<>();
+	private final Map
+		<String,
+		 ServiceTrackerMap
+			 <String, ? extends List<ServiceReferenceServiceTuple<?, ?>>>>
+				_itemClassNameInfoItemServiceTrackerMap = new HashMap<>();
 	private final Map<String, ServiceTrackerMap<String, ?>>
 		_keyedInfoItemServiceTrackerMap = new HashMap<>();
 
+	private static class ServiceReferenceServiceTupleServiceTrackerCustomizer
```

Comment:
This class seems pretty generic. Would it make sense to put it in "com.liferay.osgi.service.tracker.collections" or similar?

Reply:
we have some choices already in `ServiceTrackerCustomizerFactory.java` to use just a Function to wrap or use the ServiceWrapper one that has both the service and the properties (the `Filter` also has a method for regular maps). I did not want to hide it too much to show what it was doing under the hood.